proposals/0014-wave-size-range.md

```diff
@@ -0,0 +1,354 @@
+* Author(s): [Greg Roth](https://github.com/pow2clk)
+* Sponsor: [Greg Roth](https://github.com/pow2clk)
+* Status: **Under Consideration**
+* Planned Version: Shader Model 6.8
+
+## Introduction
+
+Shader Model 6.6 included the ability to specify a wave size on a shader entry
+ point in order to indicate either that a shader depends on or strongly prefers
+ a specific wave size.
+If that size of wave isn't available to the current hardware,
+ shader will fail to load.
+If that wave size is among those supported by the platform,
+ it must be the one used when the shader is executed.
+This enables creating custom shaders optimized for the wave size of a
+ particularly interesting platform and loading them when possible.
+Existing queries provide the developer with all the information necessary
+ to determine what a platform supports,
+ which should let them choose shaders accordingly.
+
+## Motivation
+
+Shader Model 6.6 provides no mechanism for specifying multiple wave sizes that a
+ single shader might be able to support.
+Consequently, if more than one target wave size is of interest, separate entry
+ points will have to be created for each wave size.
+These will each have to be compiled and shipped separately and, at run time,
+ the appropriate shaders will have to be selected.
+This increases the size of the shipped product and potentially slows down
+ runtime shader loading.
+
+Ranges of wave sizes are already present in areas of the D3D API.
+The query to determine what wave sizes are supported takes a range and similar
+ information is transmitted from the shader to the runtime, though not the
+ driver.
+
+### Motivational Example
+
+As an example of how this might be useful, consider a tile culling shader
+For each tile, the shader has to compute the maximum and minimum depth values.
+
+To do this without a preferred wave size involves expensive atomic an barrier
+ operations.
+To keep track of the current minimum and maximum values for each tile requires
+ a storage buffer.
+This buffer needs to be pre-initialized to unachievably low values for the
+ maximum and unachievalby high values for the minimum to ensure that whatever
+ the first value encountered is, it will replace the initialization.
+This pre-clearing requires an expensive resource barrier.
+After that setup, each thread in the tile will have to
+ use atomic operations to compare this thread's depth value
+ and determine the overall mininum and maximum depth for the whole tile,
+ storing it in the tile-specific location in the buffer.
+
+Shader Model 6.6's `WaveSize` already introduces some improvement opportunities
+ on this approach.
+A shader could specify a wave size of 64 and each thread could load 4 depth
+ values and perform a the minimum and maximum calculations locally.
+These local results are then used in a wave minimum/maximum calculation and
+ writes the final result.
+This removes the need for atomics and the advance buffer clearing.
+
+With the addition of a 
```

Comment:
Hanging incomplete sentence.

Reply:
I think you might need to refresh. This was already removed.

Reply:
Nevermind. Something is going on with my end. I don't see this locally.

Reply:
Ah right. it's also part of the removed block. Pushing commit now.